torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -34,6 +34,18 @@ void RpcAgent::setDefaultRpcAgent(std::shared_ptr<RpcAgent> defaultRpcAgent) {
   defaultRpcAgent_ = std::move(defaultRpcAgent);
 }
 
+std::unordered_map<std::string, std::string> RpcAgent::getDebugInfo() {
+  /* This would later include more info other than metrics for eg: may include
+     stack traces for the threads owned by the agent */
+  // Default implementation: return getMetrics().
+  return getMetrics();
+}
+
+void RpcAgent::addGilWaitTime(
+    const std::chrono::microseconds gilWaitTime /* unused */) {
+  // No-op default implementation
```

Comment:
We should implement this in this PR, for now we can keep track of the avg time and add it to getMetrics? We should also have some unit tests associated with this PR.

Reply:
I've implemented this with a histogram on the ThriftRpcAgent side, though for the process group agent, I guess we can implement it with just the average.